lib/libc/minimal/source/stdout/stdout_console.c

```diff
@@ -40,7 +41,18 @@ static inline int z_vrfy_zephyr_fputc(int c, FILE *stream)
 
 int fputc(int c, FILE *stream)
 {
-	return zephyr_fputc(c, stream);
```

Comment:
I'm not sure what `zephyr_fputc()` is atm (on mobile), but if it's already in `fdtable.c` then we can do proper locking there. Otherwise, it should probably be moved there.

These functions should probably be in the common libc. Otherwise every libc would need to reimplement them.

Reply:
`zephyr_fputc()` is the function installed by `__stdout_hook_install()`

https://github.com/zephyrproject-rtos/zephyr/blob/72197097aeb75ec9a7efe716d8822b0a72e7cfc5/drivers/console/uart_console.c#L66-L109

https://github.com/zephyrproject-rtos/zephyr/blob/72197097aeb75ec9a7efe716d8822b0a72e7cfc5/drivers/console/uart_console.c#L600

https://github.com/zephyrproject-rtos/zephyr/blob/72197097aeb75ec9a7efe716d8822b0a72e7cfc5/lib/libc/minimal/source/stdout/stdout_console.c#L23-L26

https://github.com/zephyrproject-rtos/zephyr/blob/72197097aeb75ec9a7efe716d8822b0a72e7cfc5/lib/libc/minimal/source/stdout/stdout_console.c#L28-L31

Reply:
This is kind of funny... it would be much easier if these different subsystems simply exposed a `fd_op_vtable` and maybe a priority in an iterable section. With that, there would be no need to install a hook at runtime.